oauthenticator/oauth2.py

```diff
@@ -980,6 +1011,20 @@ def build_access_tokens_request_params(self, handler, data=None):
             "data": data,
         }
 
+        if self.pkce:
+            # https://datatracker.ietf.org/doc/html/rfc7636#section-4.5
+            callback_handler = cast(OAuthCallbackHandler, handler)
```

Comment:
Personally I have no objection to introducing typing, but I think we should discuss it in a separate issue first.

Reply:
Oh, sorry about that. I agree that this subject should be a separate discussion.